pkg/cmd/pr/create/create.go

```diff
@@ -127,23 +145,239 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 	return cmd
 }
 
-func createRun(opts *CreateOptions) error {
+func createRun(opts *CreateOptions) (err error) {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	client := api.NewClientFromHTTP(httpClient)
+
+	ctx, err := NewCreateContext(opts)
+	if err != nil {
+		return err
+	}
+
+	var milestoneTitles []string
+	if opts.Milestone != "" {
+		milestoneTitles = []string{opts.Milestone}
+	}
+
+	state := shared.IssueMetadataState{
+		Type:       shared.PRMetadata,
+		Reviewers:  opts.Reviewers,
+		Assignees:  opts.Assignees,
+		Labels:     opts.Labels,
+		Projects:   opts.Projects,
+		Milestones: milestoneTitles,
+	}
+
+	defaultsErr := computeDefaults(*ctx, &state)
+	if defaultsErr != nil && (opts.Autofill || opts.WebMode || !opts.Interactive) {
+		return fmt.Errorf("could not compute title or body defaults: %w", defaultsErr)
+	}
+
+	if opts.TitleProvided {
+		state.Title = opts.Title
+	}
+
+	if opts.BodyProvided {
+		state.Body = opts.Body
+	}
+
+	if opts.WebMode {
+		err := handlePush(*opts, *ctx)
+		if err != nil {
+			return err
+		}
+		return previewPR(*opts, *ctx, state)
+	}
+
+	if opts.Autofill || !opts.Interactive {
+		return submitPR(*opts, *ctx, state)
```

Comment:
I may be wrong but it seems like we should have a `handlePush` call here? In this case we are calling `submitPR` without calling `handlePush` first which seems incorrect.